tensorflow/python/ops/array_ops.py

```diff
@@ -1757,6 +1757,176 @@ def _DepthToSpaceShape(op):
       [input_shape[0], height, width, new_depth])]
 
 
+def one_hot(indices, depth, on_value=None, off_value=None,
+            axis=None, dtype=None, name=None):
```

Comment:
I had a similar implementation when I first had a go at this wrapper. I think the reason @ebrevdo asked me to keep the inputs as `None` was to keep the behavior of `one_hot()` outputting a tensor of the same type as `on_value` and `off_value` (without having to re-specify it in `dtype`) . This led to all of the case checking below.

Personally, I think what you've suggested works well- it makes sense for `dtype` to act as a toggle switch rather than a type-checking device. It definitely makes the wrapper more readable too :D